ws-security-stax/src/main/java/org/apache/wss4j/stax/impl/processor/input/SecurityContextTokenInputHandler.java

```diff
@@ -56,7 +56,7 @@ public void handle(InputProcessorChain inputProcessorChain, final XMLSecurityPro
         }
 
         final QName elementName = new QName(securityContextTokenTypeJAXBElement.getName().getNamespaceURI(),
-                WSSConstants.TAG_WSC0502_IDENTIFIER.getLocalPart());
+                securityContextTokenTypeJAXBElement.getName().getLocalPart());
```

Comment:
This fix will break the next line, which gets the identifier. Really "elementName" here should be "identifierElementName". So maybe rename elementName here to identifierElementName and revert the change. Then add a new elementName for the SecurityContextToken itself.

Reply:
Yes, I see the problem. I just can't yet understand signature of `org.apache.wss4j.stax.validate.SecurityContextTokenValidator#validate` - what does identifier parameter do?

Reply:
And do I also need to support two versions of NS:
"http://schemas.xmlsoap.org/ws/2005/02/sc" and "http://docs.oasis-open.org/ws-sx/ws-secureconversation/200512" ?

Reply:
The identifier is just a way of referring to a particular SecurityContextToken. No, you don't have to support two versions - all you need to do is to make sure that the security context token element is passed through to wssSecurityProperties.getValidator.